File: 0003-longest-substring-without-repeating-characters/brute-force/brute-force.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        int max = 0;

        for (int i = 0; i < s.Length; i++)
        {
            int[] count = new int[128];

            for (int j = i; j < s.Length; j++)
            {
                if (count[s[j]] != 0)
                {
                    break;
                }
                else
                {
                    max = Math.Max(max, (j - i + 1));
                    count[s[j]] = 1;
                }
            }
        }

        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/brute-force/brute-force.c
#include <stdlib.h>
#include <string.h>

int lengthOfLongestSubstring(char* s)
{
    int length = strlen(s);
    int max    = 0;

    for (int i = 0; i < length; i++)
    {
        int count[128] = {0};

        for (int j = i; j < length; j++)
        {
            if (count[s[j]])
            {
                break;
            }
            else
            {
                int tmp = j - i + 1;
                if (tmp > max)
                {
                    max = tmp;
                }
                
                count[s[j]] = 1;
            }
        }
    }
    return max;
}